src/misc/lv_style.h

```diff
@@ -29,24 +29,22 @@ extern "C" {
 
 #define LV_STYLE_SENTINEL_VALUE     0xAABBCCDD
 
-/**
+/*
```

Comment:
Are these actually causing a warning? I've seen the `/**` convention used in many places in C code.

Reply:
The warnings were suppressed in the documentation build system. @vwheeler63 has enabled them so now all of the warnings get shown by doxygen. Some things that produce warnings and have docstrings will still get documented but other will produce warnings but the documentation doesn't populate into the final output. Ideally there should be zero warnings and zero errors even if the build completes. That is the goal. I don't have a great amount of experience with doxygen and I didn't notice that the warnings had been suppressed. This was done prior to me making any changes to the documentation build system. Someone else had done that prior to me doing any work on LVGL.

@vwheeler63 has far more experience with doxygen that I do and he found that the warnings had been suppressed. and when they were turned back on there was a HUGE number that ended up being produced.

Reply:
Got it, so the changes are guided by vwheeler63's Doxygen warnings. I see the discussion in the issue now.

Reply:
@liamHowatt @XuNeo 

Hi, Liam!

Additionally, turning Doxygen warnings ON (just the default ones so far) shows us where there where _meant to be_ documentation that was NOT making it into the eventual API documentation generated by the documentation build system, and it was hard to detect because the MEMBERS were there (e.g. macros, functions, structs, struct members), but the intended documentation strings (or "docstrings" in Python lingo) were silently not showing up!

And in what you are asking about:  "/\*\*" => "/\*" -- in that case, a string that was meant to label a GROUP of items was being erroneously interpreted (by Doxygen) as a "docstring" for the FIRST item, which of course did not make sense when it showed up only labeling the first item in the generated API documentation.  This was not a Doxygen error, but an error in formatting of the comments.  So after a discussion about this with Gabor, we settled on having Doxygen NOT bring in that string as a "docstring" since it was mis-labeling 1 item only rather than a group.  There are ways to group things in Doxygen but we reserved that for a later project.  The change from 2 "\*\*" to 1 "\*" forces Doxygen to EXCLUDE that string from the input -- intentionally and with coordination with Gabor.

Reply:
I see, thanks for the explanation! Maybe that convention came from Doxygen but people forgot that it came from there and has significance (and consequences) when used carelessly in a Doxygen context.

Reply:
> I see, thanks for the explanation! Maybe that convention came from Doxygen but people forgot that it came from there and has significance when used in a Doxygen context.

Agreed!

Reply:
Yessir that is the game plan.

I have been having a good conversation with the author of Hawkmoth which is something I am looking at as a replacement for breathe. Breathe is no longer being developed unless the author is paid to do it. So it needs to be replaced with something that is still being developed. The thing I like about Hawkmoth is it will parse doxygen docstrings without using doxygen. It uses clang and attaches to the compiler using it's C API. This is done using ctypes in Python. The other thing I really like about it is not only is it able to read doxygen docstrings but it will also read sphinx formatted ReST docstrings as well. Nice thing about this is we will have the ability to create connections between the API documentation and things located in the sphinx documentation which is something that is not able to currently be done. It also allows for documented sections in the C header files. So things like the widget documentation that is currently located in the docs folder is able to be relocated into the C header files for each widget. I can organize things however it is wanting to be done but the big thing is the documentation is able to be located with the code instead of being a separate file that needs to be updated.

Reply:
My point to that last comment is in order for it to work properly the doxygen syntax needs to be correct otherwise it is going to have issues.

Reply:
> the big thing is the documentation is able to be located with the code

I think this is important as well -- without it, it makes it that much harder to learn about how to use LVGL.

Reply:
@liamHowatt 

Hi, Liam!

In this context, is it you or me who is responsible to press the [Resolve Conversation] button in the "Files" tab?

--Vic

Reply:
There is no hard rule. I personally only resolve conversations if I started the conversation, the issue is resolved, and there isn't an ongoing discussion. If someone decides to resolve it sooner that's fine.

Reply:
> There is no hard rule. I personally only resolve conversations if I started the conversation, the issue is resolved, and there isn't an ongoing discussion. If someone decides to resolve it sooner that's fine.

Cool.  Thanks!  :-)